src/Uno.UWP/Gaming/Input/Gamepad.Android.cs

```diff
@@ -0,0 +1,298 @@
+				Keycode.ButtonB => GamepadButtons.B,
+				Keycode.ButtonY => GamepadButtons.Y,
+				Keycode.DpadUp => GamepadButtons.DPadUp,
+				Keycode.DpadRight => GamepadButtons.DPadRight,
+				Keycode.DpadDown => GamepadButtons.DPadDown,
+				Keycode.DpadLeft => GamepadButtons.DPadLeft,
+				Keycode.DpadUpLeft => GamepadButtons.DPadUp | GamepadButtons.DPadLeft,
+				Keycode.DpadUpRight => GamepadButtons.DPadUp | GamepadButtons.DPadRight,
+				Keycode.DpadDownRight => GamepadButtons.DPadDown | GamepadButtons.DPadRight,
+				Keycode.DpadDownLeft => GamepadButtons.DPadDown | GamepadButtons.DPadLeft,
+				Keycode.ButtonThumbl => GamepadButtons.LeftThumbstick,
+				Keycode.ButtonThumbr => GamepadButtons.RightThumbstick,
+				Keycode.ButtonL1 => GamepadButtons.LeftShoulder,
+				Keycode.ButtonR1 => GamepadButtons.RightShoulder,
+				Keycode.ButtonStart => GamepadButtons.Menu,
+				Keycode.ButtonSelect => GamepadButtons.View,
+				_ => GamepadButtons.None,
+			};
+
+		private static IReadOnlyList<Gamepad> GetGamepadsInternal()
+		{
+			var cachedDeviceIds = _gamepadCache.Keys.ToArray();
+			var connectedIds = InputDevice.GetDeviceIds() ?? Array.Empty<int>();
+
+			//remove disconnected
+			var disconnectedDevices = cachedDeviceIds.Except(connectedIds);
+			_gamepadCache.RemoveKeys(disconnectedDevices);
+
+			//add newly connected
+			foreach (var deviceId in connectedIds)
+			{
+				if (TryGetOrCreateGamepad(deviceId, out var gamepad))
+				{
+					if (!_gamepadCache.ContainsKey(deviceId))
+					{
+						_gamepadCache.Add(deviceId, gamepad!);
+					}
+				}
+			}
+
+			return _gamepadCache.Values.OrderBy(g => g._nativeDeviceId).ToArray();
+		}
+
+		private static void EnsureInputManagerInitialized()
+		{
+			if (_inputManager == null)
+			{
+				_inputManager = (InputManager?)ContextHelper.Current.GetSystemService(Context.InputService);
+
+				if (_inputManager == null)
+				{
+					throw new InvalidOperationException("Cannot access input manager");
+				}
+			}
+		}
+
+		private static void StartGamepadAdded() =>
+			AttachInputDeviceListener();
+
+		private static void EndGamepadAdded() =>
+			DetachInputDeviceListener();
+
+		private static void StartGamepadRemoved() =>
+			AttachInputDeviceListener();
+
+		private static void EndGamepadRemoved() =>
+			DetachInputDeviceListener();
+
+		private static void AttachInputDeviceListener()
+		{
+			if (_listener != null)
+			{
+				return;
+			}
+
+			EnsureInputManagerInitialized();
+
+			_listener = new InputDeviceListener();
+			_inputManager!.RegisterInputDeviceListener(_listener, null);
+		}
+
+		private static void DetachInputDeviceListener()
+		{
+			if (_gamepadAddedWrapper.HasSubscribers ||
+				_gamepadRemovedWrapper.HasSubscribers)
+			{
+				return;
+			}
+
+			_inputManager!.UnregisterInputDeviceListener(_listener);
+			_listener?.Dispose();
+			_listener = null;
+		}
+
+		private static bool IsGamepad(InputDevice? inputDevice) =>
+			inputDevice?.Sources.HasFlag(InputSourceType.Gamepad) == true;
```

Comment:
@jeromelaban this is a bit problematic, because a gamepad can be connected anytime (or could already be connected before the app was started), so we may receive a key event or generic motion event from it at any point in time 🤔 . Do you think this could cause performance issues?

Reply:
Technically this should not be too costly, as `Sources` is just a enum property.